Assets/Scripts/Escape/GeometricEscapeDetector.cs

```diff
@@ -0,0 +1,35 @@
+using UnityEngine;
+
+// Geometric escape detector.
+//
+// The geometric escape detector checks whether the agent is between the threat and its target.
+public class GeometricEscapeDetector : EscapeDetectorBase {
+  // The range buffer factor allows the interceptor to be slightly further away from the target than
+  // the target is to its target before declaring an escape.
+  private const float _rangeBufferFactor = 1.1f;
+
+  public GeometricEscapeDetector(IAgent agent) : base(agent) {}
+
+  public override bool IsEscaping(IHierarchical target) {
+    if (target == null) {
+      return false;
+    }
+
+    Transformation agentToTargetTransformation = Agent.GetRelativeTransformation(target);
+
+    // Check if the target has a target of its own.
+    if (target.Target != null && !target.Target.IsTerminated) {
+      Vector3 targetRelativePositionToTarget = target.Target.Position - target.Position;
+      // The target is escaping if the distance from the agent to the target is greater than the
+      // distance from the target to its target or if the agent is geometrically behind the target
+      // relative the target's own target.
+      return agentToTargetTransformation.Position.Range >
+                 targetRelativePositionToTarget.magnitude * _rangeBufferFactor ||
+             Vector3.Dot(-agentToTargetTransformation.Position.Cartesian,
+                         targetRelativePositionToTarget) < 0;
+    }
+
```

Comment:
There's kind of an inconsistency in behavior here where both detectors fall back to checking `agentToTargetTransformation.Velocity.Range < 0` when the target has no target, but they check this condition differently.

The geometric escape detector can declare escape even with positive closing velocity if the agent is geometrically behind, but the time escape detector doesn't have this opportunity/behavior? Maybe this is intentional? Could use explanation in comments maybe

Reply:
Yes, this is intentional.  The time escape detector only cares about the time (at least this was the thought in my mind).  I don't think any of these escape detectors are necessarily optimal, just ideas that I had.